lib/s3.py

```diff
@@ -50,11 +51,18 @@ def get_key(hostname):
     return None
 
 
-def is_key_present(url):
+def is_key_present(url: urllib.parse.ParseResult):
+    """Checks if an S3 key is available for the given url"""
     return get_key(url.hostname) is not None
 
 
-def sign_request(url, method='GET', checksum=SHA256_NIL, headers={}, region='any', service='S3'):
+def sign_request(url: urllib.parse.ParseResult, method='GET', checksum=SHA256_NIL, headers={}) -> Dict[str, str]:
```

Comment:
Please squash the "s3: annotate and document the APIs" commit. It looks rather odd otherwise.  Also, changing "S3" to "s3" seems unrelated.

Reply:
Sure.

To be honest, I feel like Linode is completely ignoring both the region and the service values.  I changed this to "s3" just because I found an official example which contained it as lowercase...